middleware/locale/locale.go

```diff
@@ -0,0 +1,56 @@
+package locale
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/mholt/caddy/middleware"
+	"github.com/mholt/caddy/middleware/locale/method"
+)
+
+// Locale is a middleware to detect the user's locale.
+type Locale struct {
+	Next     middleware.Handler
+	Locales  []string
+	Methods  []method.Method
+	Settings *method.Settings
+}
+
+// ServeHTTP implements the middleware.Handler interface.
+func (l *Locale) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
+	candidates := []string{}
+	for _, method := range l.Methods {
+		candidates = append(candidates, method(r, l.Settings)...)
+	}
+
+	locale := l.firstValid(candidates)
```

Comment:
»First valid« *what*? It's not apparent here, and still unclear when reading function ```firstValid```. I mean, you have already thrown out any invalid configurations in ```SetupLocale```, so what gives?